app/views/finders/show.html.erb

```diff
@@ -59,19 +59,35 @@
 </header>
 
 <div class="filtering">
-  <%= render finder.facets %>
-  <div class='js-live-search-results-block'>
-    <div class="filtered-results">
-      <div aria-live='assertive' id='js-search-results-info'>
-        <% if finder.show_generic_description? %>
-          <%= render_mustache('result_count_generic', @results.to_hash) %>
-        <% else %>
-          <%= render_mustache('result_count', @results.to_hash) %>
-        <% end %>
+  <form method="get" action="<%= finder.slug %>" class="js-live-search-form">
+    <%= render finder.facets %>
+
+    <% if finder.sort.present? %>
+      <div class="filter-form">
+        <div class="govuk-form-group">
+          <%= label_tag 'order', 'Sort by', class: 'govuk-label' %>
+          <%= select_tag 'order', finder.sort_options,
+                         class: 'govuk-select js-order-results',
+                         'aria-controls': 'js-search-results-info',
+                         data: { 'default-sort-option' => finder.default_sort_option_value,
+                                 'relevance-sort-option' => finder.relevance_sort_option_value } %>
```

Comment:
Perhaps you could provide a `sort_options` method containing this hash, then you could expose a simpler sorting api in finder presenter, and perhaps introduce a separate class that handles sorting logic? This would simplify the finder presenter and views and would make it easier to skim read the sorting logic.